reclass/storage/yaml_fs/__init__.py

```diff
@@ -50,12 +50,16 @@ def _path_mangler_inner(path):
 
 class ExternalNodeStorage(NodeStorageBase):
 
-    def __init__(self, nodes_uri, classes_uri):
+    def __init__(self, nodes_uri, classes_uri, add_subdir_to_node):
         super(ExternalNodeStorage, self).__init__(STORAGE_NAME)
 
         if nodes_uri is not None:
             self._nodes_uri = nodes_uri
-            self._nodes = self._enumerate_inventory(nodes_uri, NameMangler.nodes)
+            if add_subdir_to_node:
+                self._nodes = self._enumerate_inventory(nodes_uri, NameMangler.classes)
+            else:
+                self._nodes = self._enumerate_inventory(nodes_uri, NameMangler.nodes)
+
```

Comment:
This is the main change of this MR

if `add_subdir_to_node` is true, then we use the same `NameMangler` used for classes that concatenates all subdirs with `.` and add filename at the end:

example:

```
inventory/nodes/prod/mysql.yaml
```

returns

```
prod.mysql
```